armory/utils/metrics.py

```diff
@@ -159,6 +159,39 @@ def _snr_spectrogram(x_i, x_adv_i):
     return signal_power / noise_power
 
 
+def word_error_rate(y, y_pred):
+    """
+    Return the word error rate for a batch of transcriptions.
+    """
+    if len(y) != len(y_pred):
+        raise ValueError(f"len(y) {len(y)} != len(y_pred) {len(y_pred)}")
+    return [_word_error_rate(y_i, y_pred_i) for (y_i, y_pred_i) in zip(y, y_pred)]
```

Comment:
FYI WER on a test dataset is typically computed by summing the total number of insertions, deletions, and substitutions and dividing it by the total number of words, rather than taking a mean of individual utterance WERs. The numbers will be different and you might not match the results reported by others. 

E.g. in deepspeech.pytorch they are doing it the way I described (the `total_wer` variable is actually the raw edit distance, and not real WER): https://github.com/SeanNaren/deepspeech.pytorch/blob/master/deepspeech_pytorch/testing.py#L92

Reply:
I see. To clarify, it should be the raw edit distance of the entire dataset divided by the number of words in the entire dataset, correct?

Reply:
That's right.